packages/liveblocks-react/src/room.tsx

```diff
@@ -1477,54 +1498,74 @@ function useEditThreadMetadata<M extends BaseMetadata>() {
  */
 function useCreateComment(): (options: CreateCommentOptions) => CommentData {
   const client = useClient();
-  const room = useRoom();
+
   return React.useCallback(
     ({ threadId, body, attachments }: CreateCommentOptions): CommentData => {
       const commentId = createCommentId();
       const createdAt = new Date();
 
+      const userId = getCurrentUserId(client);
+
+      const { store, onMutationFailure } = getRoomExtrasForClient(client);
+
+      const existing = store
+        .getFullState()
+        .threadsDB.getEvenIfDeleted(threadId);
+      if (existing === undefined) {
+        throw new Error(
+          `Cannot create comment in thread "${threadId}" because the thread does not exist in cache.`
+        );
+      }
```

Comment:
@marcbouchenoire @nvie 

We now throw an error if the thread id passed to the function returned by `useCreateComment` (or `useEditComment`, `useEditThreadMetadata`, `useAddReaction`, etc.) doesn't map to a thread in cache. 

Also, to be consistent with the old behavior, if user id is not defined, we set the user id to `anonymous`